packages/codegen-ui-react/lib/forms/form-renderer-helper/render-checkers.ts

```diff
@@ -23,8 +22,13 @@ export const isModelDataType = (
 ): config is FieldConfigMetadata & { dataType: { model: string } } =>
   !!(config.dataType && typeof config.dataType === 'object' && 'model' in config.dataType);
 
+export const isEnumDataType = (
+  config: FieldConfigMetadata,
+): config is FieldConfigMetadata & { dataType: { enum: string } } =>
+  !!(config.dataType && typeof config.dataType === 'object' && 'enum' in config.dataType);
```

Comment:
prefer to check that `enum` is not undefined rather than just stopping at the property existing

Reply:
Finally addressing here - https://github.com/aws-amplify/amplify-codegen-ui/pull/991